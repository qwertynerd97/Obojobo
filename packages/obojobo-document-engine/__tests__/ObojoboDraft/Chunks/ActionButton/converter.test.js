import Converter from 'ObojoboDraft/Chunks/ActionButton/converter'

describe('ActionButton Converter', () => {
	test('slateToObo converts a Slate node', () => {
		const slateNode = {
			key: 'mockKey',
			type: 'mockType',
			data: {
				get: () => {
					return {
						actions: [
							{
								type: 'mockType',
								value: ''
							},
							{
								type: 'mockOtherType',
								value: '{"id": "mockId"}'
							}
						]
					}
				}
			}
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			content: { label: 'mockLabel' }
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node using a textGroup', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			content: { textGroup: [{ text: { value: 'mockLabel' } }] }
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node with triggers', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			content: {
				label: 'mockLabel',
				triggers: [
					{
						actions: [{ type: 'mockType' }, { type: 'mockOtherType', value: { id: 'mockId' } }]
					}
				]
			}
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})
})
