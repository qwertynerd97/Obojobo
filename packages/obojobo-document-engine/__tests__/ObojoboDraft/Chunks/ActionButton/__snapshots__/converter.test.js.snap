// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton Converter oboToSlate converts an OboNode to a Slate node 1`] = `
Object {
  "data": Object {
    "content": Object {
      "actions": Array [],
      "label": "mockLabel",
    },
  },
  "key": "mockKey",
  "object": "block",
  "type": "mockType",
}
`;

exports[`ActionButton Converter oboToSlate converts an OboNode to a Slate node using a textGroup 1`] = `
Object {
  "data": Object {
    "content": Object {
      "actions": Array [],
      "label": "mockLabel",
    },
  },
  "key": "mockKey",
  "object": "block",
  "type": "mockType",
}
`;

exports[`ActionButton Converter oboToSlate converts an OboNode to a Slate node with triggers 1`] = `
Object {
  "data": Object {
    "content": Object {
      "actions": Array [
        Object {
          "type": "mockType",
          "value": "",
        },
        Object {
          "type": "mockOtherType",
          "value": "{\\"id\\":\\"mockId\\"}",
        },
      ],
      "label": "mockLabel",
    },
  },
  "key": "mockKey",
  "object": "block",
  "type": "mockType",
}
`;

exports[`ActionButton Converter slateToObo converts a Slate node 1`] = `
Object {
  "children": Array [],
  "content": Object {
    "label": "",
    "triggers": Array [
      Object {
        "actions": Array [
          Object {
            "type": "mockType",
            "value": Object {},
          },
          Object {
            "type": "mockOtherType",
            "value": Object {
              "id": "mockId",
            },
          },
        ],
        "type": "onClick",
      },
    ],
  },
  "id": "mockKey",
  "type": "mockType",
}
`;
